### PR TITLE
bpo-47120: make POP_JUMP_IF_NONE/NOT_NONE relative

### PR DESCRIPTION
**Pull request by iritkatriel**
https://bugs.python.org/issue47120

### CODE REVIEW

**Comment by iritkatriel** [comment]
The case of POP_JUMP_IF_TRUE/IF_FALSE is more complicated because of the COMPARE_OP specialisations. I suggest we do it separately (my current patch doesn't work unless I turn off specialization, debugging).

File: Misc/NEWS.d/next/Core and Builtins/2022-04-06-22-50-31.bpo-47120.mbfHs5.rst [hunk 1]
```diff
@@ -0,0 +1 @@
+Make :opcode:`POP_JUMP_IF_TRUE` and :opcode:`POP_JUMP_IF_FALSE` virtual, mapping to new relative jump opcodes.
```
Comment (iritkatriel) [inline Misc/NEWS.d/next/Core and Builtins/2022-04-06-22-50-31.bpo-47120.mbfHs5.rst]:
```suggestion
Make :opcode:`POP_JUMP_IF_NONE` and :opcode:`POP_JUMP_IF_NOT_NONE` virtual, mapping to new relative jump opcodes.
```

**Comment by markshannon** [comment]
The `FORWARD` jumps don't need to include `CHECK_EVAL_BREAKER()`, so I think you'll need to implement each instruction independently and not use `JUMP_TO_INSTRUCTION`.

**Comment by markshannon** [comment]
Which means the forwards jumps can be made simpler
```
TARGET(POP_JUMP_FORWARD_IF_NOT_NONE) {
    PyObject *value = POP();
    if (value != Py_None) {
        JUMPBY(oparg);
    }
    Py_DECREF(value);
    DISPATCH();
}

**Review by markshannon — APPROVED** [review]
:+1:

**Comment by iritkatriel** [comment]
This will be merged as part of https://github.com/python/cpython/pull/32400

**Pull request closed without merging**